include/net/cloud.h

```diff
@@ -62,6 +62,27 @@ enum cloud_endpoint_type {
 	CLOUD_EP_PRIV_END = INT16_MAX
 };
 
+/**@brief Cloud connect results. */
+enum cloud_connect_result {
+	CLOUD_CONNECT_RES_SUCCESS = 0,
+
+	CLOUD_CONNECT_RES_ERR_NOT_INITD = -1,
+	CLOUD_CONNECT_RES_ERR_NETWORK = -2,
+	CLOUD_CONNECT_RES_ERR_BACKEND = -3,
+	CLOUD_CONNECT_RES_ERR_MISC = -4,
+
+	/* keep existing known error numbers */
```

Comment:
This will not work, as the number will be different for different libc implementations (Zephyr uses two - minimal libc and newlib, and they differ on numeric values of specific errnos and both are used across NCS). I'd just use some consecutive values here and do a proper errno translation.

Reply:
the 4 errors below are from the mqtt_connect result (from bsd lib?).
i could not locate the actual error code defines (for -95, -111, -116).

Reply:
Depending on which libc implementation is used, these error codes are defined either [here (minimal libc)](https://github.com/zephyrproject-rtos/zephyr/blob/master/lib/libc/minimal/include/errno.h) or [here (newlib)](https://github.com/eblot/newlib/blob/master/newlib/libc/include/sys/errno.h). As you can see the actual numbers differ between implementations.

Reply:
thanks, that newlib link was what i needed.